libraries/AP_Baro/AP_Baro_LPS25H.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "AP_Baro_LPS25H.h"
+#include <unistd.h>
+#include <utility>
+#include <stdio.h>
+extern const AP_HAL::HAL &hal;
+
+
+#define LPS25H_ID            	  0xBD
+#define LPS25H_REG_ID			  0x0F
+#define LPS25H_CTRL_REG1_ADDR     0x20
+#define LPS25H_CTRL_REG2_ADDR     0x21
+#define LPS25H_CTRL_REG3_ADDR     0x22
+#define LPS25H_CTRL_REG4_ADDR     0x23
+#define LPS25H_FIFO_CTRL          0x2e
+#define LPS25H_TEMP_OUT_ADDR      (uint8_t)0x2B
+#define PRESS_OUT_XL_ADDR		  0x28
+
+
+AP_Baro_LPS25H::AP_Baro_LPS25H(AP_Baro &baro, AP_HAL::OwnPtr<AP_HAL::Device> dev)
+    : AP_Baro_Backend(baro)
+    , _dev(std::move(dev))
+{
+}
+
+AP_Baro_Backend *AP_Baro_LPS25H::probe(AP_Baro &baro,
+                                       AP_HAL::OwnPtr<AP_HAL::Device> dev)
+{
+    if (!dev) {
+        return nullptr;
+    }
+    AP_Baro_LPS25H *sensor = new AP_Baro_LPS25H(baro, std::move(dev));
+    if (!sensor || !sensor->_init()) {
+        delete sensor;
+        return nullptr;
+    }
+    return sensor;
+}
+
+bool AP_Baro_LPS25H::_init()
+{
+    if (!_dev | !_dev->get_semaphore()->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+        return false;
+    }
+    _has_sample = false;
+
+    _dev->set_speed(AP_HAL::Device::SPEED_HIGH);
+
+    uint8_t whoami;
+    if (!_dev->read_registers(LPS25H_REG_ID, &whoami, 1)  ||
+        whoami != LPS25H_ID) {
+        // not a LPS25H
+        _dev->get_semaphore()->give();
+        return false;
+    }
+
+	//init control registers.
+	_dev->write_register(LPS25H_CTRL_REG1_ADDR,0x00); // turn off for config 
+	_dev->write_register(LPS25H_CTRL_REG2_ADDR,0x40); 
+	_dev->write_register(LPS25H_FIFO_CTRL, 0x01);
```

Comment:
If I'm getting this right, you are enabling FIFO mode but then putting it in bypass mode, is there any advantage to this?

Reply:
originally thought FIFO has to be enabled in order to use Bypass mode because it uses the first address of the FIFO. 
disabled it in my last commit